### PR TITLE
Add support for gzip/snappy compression to produce requests

### PR DESCRIPTION
**Pull request by jedisct1**
This adds support for gzip and snappy compression when sending messages.

`snappy.rs` and `gzip.rs` should probably be part of the `compression` module, but I'll submit a change for that in another pull request in order to keep that one diff-readable.

### CODE REVIEW

File: src/protocol.rs [hunk 1]
```diff
@@ -1306,8 +1345,18 @@ impl ToByte for Message {
         } else {
             try!(self.key.encode(&mut buf));
         }
-
-        try!(self.value.encode(&mut buf));
+        match self.attributes & 3 {
```
Comment (spicavigo) [inline src/protocol.rs]: What does 3 signify here? Please provide a comment, or better yet, a Constant name

**Comment by spicavigo** [comment]
This looks good. However, I have not tested it on my machine yet. Could you please confirm that sending  of snappy msgs works?

I think I will be working on setting up some more automated tests, specifically ones with actual Kafka servers.

**Comment by jedisct1** [comment]
We've been using this in production for the past two weeks without any issues so far.

But I agree that we should definitely add some automated tests.

**Comment by spicavigo** [comment]
+1

Merging it now!

**Pull request merged**